app/scripts/controllers/detect-tokens.js

```diff
@@ -24,36 +23,55 @@ export default class DetectTokensController {
     preferences,
     network,
     keyringMemStore,
+    tokenList,
   } = {}) {
     this.preferences = preferences;
     this.interval = interval;
     this.network = network;
     this.keyringMemStore = keyringMemStore;
+    this.tokenList = tokenList;
+  }
+
+  async _getTokenBalances(tokens) {
+    const ethContract = this.web3.eth
+      .contract(SINGLE_CALL_BALANCES_ABI)
+      .at(SINGLE_CALL_BALANCES_ADDRESS);
+    return new Promise((resolve, reject) => {
+      ethContract.balances([this.selectedAddress], tokens, (error, result) => {
+        if (error) {
+          return reject(error);
+        }
+        return resolve(result);
+      });
+    });
   }
 
   /**
-   * For each token in @metamask/contract-metadata, find check selectedAddress balance.
+   * For each token in tokenlist ptovided by the TokenListController, find check selectedAddress balance.
    */
   async detectNewTokens() {
     if (!this.isActive) {
       return;
     }
-    if (this._network.store.getState().provider.chainId !== MAINNET_CHAIN_ID) {
+    console.log(this._tokenList.state.tokenList);
+    if (
+      this._network.store.getState().provider.chainId !== MAINNET_CHAIN_ID &&
+      this._network.store.getState().provider.chainId !== '0x38'
```

Comment:
I might have asked this question elsewhere: are we only supporting token detection on mainnet and binance to start?

Reply:
Also, probably helpful to replace `'0x38'` with a descriptive constant like `MAINNET_CHAIN_ID`

Reply:
I am unsure of that so I changed that to check if the `tokenList` is empty.